### PR TITLE
Simplify generators

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This is the first step in refactoring the test generators. The aim is to make it really, really simple. In this PR, we default to using all the non-default properties as input properties. In a future PR I would like to get rid of the virtual methods as much as possible. Why not use constructor parameters instead, or otherwise use properties that can be set in the constructor?

I've also used used @robkeim's suggestion of losing the "Exercise" postfix for the exercises.

### CODE REVIEW

File: generators/Exercises/Acronym.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Generators.Exercises
+{
+    public class Acronym : EqualityExercise
+    {
+        public Acronym() : base("acronym")
```
Comment (robkeim) [inline generators/Exercises/Acronym.cs]: Another potential simplification would be to generate the exercise name from the Class name using reflection. The rules are pretty simple:
1. Acronym -> acronym
2. NthPrime -> nth-prime

You could argue it actually makes the code more obscure but it's some food for thought :)

Comment (ErikSchierboom) [inline generators/Exercises/Acronym.cs]: I like it! The PR has been updated. Don't merge it yet, as there is still a problem with the multi-line formatting in `beer-song`. I think that's the next thing I want to get right: figuring out how to best configure how the output is formatted.

File: generators/Exercises/Exercise.cs [hunk 2]
```diff
@@ -42,40 +42,24 @@ private static void AddTestMethodUsingNamespaces(TestClass testClass)
...
 
-        protected virtual TestMethodData CreateTestMethodData(CanonicalData canonicalData, CanonicalDataCase canonicalDataCase, int index) 
-            => new TestMethodData
-                {
-                    CanonicalData = canonicalData,
-                    CanonicalDataCase = canonicalDataCase,
-                    Index = index,
-                    Options = CreateTestMethodOptions(canonicalData, canonicalDataCase, index)
-                };
+            if (testMethodData.Options.TestedMethodType == TestedMethodType.BooleanComparison)
```
Comment (robkeim) [inline generators/Exercises/Exercise.cs]: Minor: To future proof this you may want to have a switch with the two types existing today and then throw an exception for the default. That way if we need to add a new type in the future it's clear where it needs to be plumbed through to.

File: generators/Output/BooleanTestMethodGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        protected override IEnumerable<string> Body
+        {
+            get
+            {
+                switch (TestMethodData.Options.TestedMethodFormat)
+                {
+                    case TestedMethodFormat.Static:
+                        return new[] { $"{Assertion}({TestedClassName}.{TestedMethodName}({Input}));" };
+                    case TestedMethodFormat.Instance:
+                        return new[] { $"var sut = new {TestedClassName}();", $"{Assertion}(sut.{TestedMethodName}({Input}));" };
```
Comment (robkeim) [inline generators/Output/BooleanTestMethodGenerator.cs]: What is `sut`?

File: generators/Output/EqualityTestMethodGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        protected virtual string FormattedExpectedVariable
+        {
+            get
+            {
+                var lines = ExpectedToMultiLineString(Expected).Split('\n');
+                if (lines.Length == 1)
+                    return lines[0];
+
+                return "\n" + string.Join("\n", lines.Select(line => $"{line}"));
```
Comment (robkeim) [inline generators/Output/EqualityTestMethodGenerator.cs]: What's the difference between what you have an `line => line`?

File: generators/Output/TestedMethodFormat.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+namespace Generators.Output
+{
+    public enum TestedMethodFormat
+    {
+        Static = 0,
```
Comment (robkeim) [inline generators/Output/TestedMethodFormat.cs]: Minor: no need to start with 0 as it's the default

Comment (robkeim) [inline generators/Output/TestedMethodFormat.cs]: Same below

File: generators/Exercises/Exercise.cs (hunk 2 above)
Comment (ErikSchierboom) [inline generators/Exercises/Exercise.cs]: I like this!

File: generators/Output/BooleanTestMethodGenerator.cs (hunk 3 above)
Comment (ErikSchierboom) [inline generators/Output/BooleanTestMethodGenerator.cs]: It stands for "system under test" and is common jargon in some test frameworks.

File: generators/Output/EqualityTestMethodGenerator.cs (hunk 4 above)
Comment (ErikSchierboom) [inline generators/Output/EqualityTestMethodGenerator.cs]: Ah, it used to have tabs! I'll fix it.

File: generators/Output/BooleanTestMethodGenerator.cs (hunk 3 above)
Comment (robkeim) [inline generators/Output/BooleanTestMethodGenerator.cs]: Nice, you learn something new everyday!

**Comment by ErikSchierboom** [comment]
@robkeim I've made some further simplifications that I think turned out quite nice :) There are two major open points left:

- Better handle large expected values (by introducing a variable)
- Better handle large strings (by concatenating them over multiple lines)
- Fix the failing tests (which are due to namespace issues)

**Comment by ErikSchierboom** [comment]
@robkeim I've updated the PR and I am happy with the end result. Let me know how you feel!

**Comment by robkeim** [comment]
Excellent work @ErikSchierboom, thanks for the hard work!

**Pull request merged**